src/index.js

```diff
@@ -36,12 +35,23 @@ class FloodSub extends BaseProtocol {
    * @constructor
    */
   constructor (peerInfo, registrar, options = {}) {
-    assert(PeerInfo.isPeerInfo(peerInfo), 'peer info must be an instance of `peer-info`')
+    if (!PeerInfo.isPeerInfo(peerInfo)) {
+      throw new Error('peer info must be an instance of `peer-info`')
+    }
 
     // registrar handling
-    assert(registrar && typeof registrar.handle === 'function', 'a handle function must be provided in registrar')
-    assert(registrar && typeof registrar.register === 'function', 'a register function must be provided in registrar')
-    assert(registrar && typeof registrar.unregister === 'function', 'a unregister function must be provided in registrar')
+    if (typeof registrar !== 'object') {
+      throw new Error('a registrar object is required')
+    }
+    if (typeof registrar.handle !== 'function') {
```

Comment:
Can we have a function to validate registrar? This way, we would not have the constructor so noisy

Reply:
Done!